src/inline.jl

```diff
@@ -88,14 +88,36 @@ function display(d::InlineDisplay, M::MIME, x)
                               "data" => d)))
 end
 
+
+
+# extract width and height from a PNG file header inside a base64 string
+function png_wh(img::String)
+    # PNG header is 8 bytes, 4 byte chunk size, 4 byte IHDR string, 8 bytes for w, h
+    decoded = base64decode(img[1:32])  # Base64 encodes 6 bits per character
+    if any(decoded[13:16] .!= b"IHDR")  # check if the header looks reasonable
+        throw(ArgumentError("Base64-encoded PNG has a badly formed header."))
```

Comment:
I don't think we should throw an error here; we should let the front end decide what to do with malformed PNG data (i.e. just don't set any metadata).

Reply:
Makes sense, I'll make this change.